### PR TITLE
MNT: move tests out of src and into tests/

### PR DESCRIPTION
**Pull request by akrherz**
#### Description Of Changes

Moves tests and vcr cassettes into top level `tests/` directory.

#### Checklist

- [x] refs #301
- [x] Tests added
- [x] Fully documented

### CODE REVIEW

**Comment by akrherz** [comment]
This is green, but I am not fully certain about the second commit added to this PR.

File: .github/workflows/tests-pypi.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
 
     - name: Run tests
       run: |
-        python -m coverage run -p -m pytest src/siphon
+        python -m coverage run -p -m pytest
```
Comment (dopplershift) [inline .github/workflows/tests-pypi.yml:80]:
```suggestion
        python -m coverage run -p -m pytest tests/
```

File: .github/workflows/tests-conda.yml [hunk 2]
```diff
@@ -69,7 +69,7 @@ jobs:
       # By running coverage in "parallel" mode and "combining", we can clean up the path names
       run: |
         export TEST_DATA_DIR=$GITHUB_WORKSPACE/staticdata
-        python -m coverage run -p -m pytest src/siphon
+        python -m coverage run -p -m pytest
```
Comment (dopplershift) [inline .github/workflows/tests-conda.yml:72]:
```suggestion
        python -m coverage run -p -m pytest tests/
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
While I think the current state works, this points `pytest` directly to the tests.

**Review by dopplershift — APPROVED** [review]

**Comment by dopplershift** [comment]
Thanks @akrherz ! 🎉

**Pull request merged**